pwiz_tools/Skyline/Model/Lib/EncylopeDiaLibrary.cs

```diff
@@ -159,9 +158,61 @@ public override LibraryDetails LibraryDetails
         {
             get
             {
+                var detailsByFileName = new Dictionary<string, SpectrumSourceFileDetails>();
+
+                try
+                {
+                    lock (_pooledSqliteConnection)
+                    {
+                        using (var select = new SQLiteCommand(_pooledSqliteConnection.Connection))
+                        {
+                            // ReSharper disable LocalizableElement
+
+                            // Query for the source files detail information.
+                            select.CommandText = @"select one.SourceFile, BestSpectra, MatchedSpectra
+from (
+  select SourceFile, count(*) as MatchedSpectra
+  from entries as s
+  group by SourceFile) as one
+inner join (
+  select SourceFile, count(*) as BestSpectra
+  from entries as e
+  where Score = (select min(Score) from entries where e.PeptideModSeq = PeptideModSeq)

```

Comment:
I think you also need "e.Charge = Charge" in the WHERE clause here.
The compound key that uniquely identifies a row in the "entries" table is "PeptideModSeq, Charge, SourceFile".

This query also looks like it would be very slow because there are no indexes in the .elib. It sounded like maybe you had convinced Brian to add some indexes but we still need to handle older .elib files.

Also, Score = (select min(Score)...) means that there could be more than one best spectrum for a given peptide if there is a tie for the lowest score. Wouldn't we want to say that if there is a tie for the best score, the best spectrum is the one that we found first, or something, so that you don't end up with more best spectra than peptides?

(I'm thinking this query should be changed to be something that reads all the necessary rows from the tables and puts them in some dictionaries and we count things using C# code)